File: 0045-jump-game-ii/0045-jump-game-ii.py
class Solution:
    def jump(self, nums: List[int]) -> int:
        
        n = len(nums)
        res = 0
        l = r = 0
        while r < n-1:
            temp = r
            for j in range(l, r+1):
                r = max(r, j + nums[j])
            l = temp + 1
            res += 1    
        
        return res
            
            
        

File: 0045-jump-game-ii/0045-jump-game-ii.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        
        int n = nums.size();
        int l=0, r=0, res=0;
        while (r<n-1){
            int temp = r;
            
            for (int j = l; j <= temp; j++){r = max(r, j + nums[j]);}
            
            l = temp + 1;
            res += 1;
        }
        
        return res;   
        
    }
};